torch/nn/parallel/distributed.py

```diff
@@ -657,7 +660,19 @@ def forward(self, *inputs, **kwargs):
             self.reducer._set_forward_pass_work_handle(
                 work, self.ddp_uneven_inputs_config.ddp_join_divide_by_initial_world_size
             )
-
+            if self.ddp_uneven_inputs_config.ddp_join_throw_on_early_termination:
+                # Check if need to throw StopIteration.
+                zeros = torch.zeros(1, device=self.device)
+                # TODO: This allreduce can be scheduled and waited in the C++
+                # backwards pass only if the above allreduce indicated at least
+                # one rank terminated. This will be more efficient than blocking
+                # in the forward pass.
```

Comment:
If we move this to the end of the forward pass, models with layers like SyncBatchNorm would get stuck?